server/src/main/java/org/cloudfoundry/identity/uaa/home/HomeController.java

```diff
@@ -12,11 +12,21 @@
  *******************************************************************************/
 package org.cloudfoundry.identity.uaa.home;
 
+import static org.springframework.util.StringUtils.hasText;
+
+import java.net.URISyntaxException;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import org.apache.http.client.utils.URIBuilder;
 import org.cloudfoundry.identity.uaa.client.ClientMetadata;
-import org.cloudfoundry.identity.uaa.client.JdbcClientMetadataProvisioning;
+import org.cloudfoundry.identity.uaa.client.ClientMetadataProvisioning;
```

Comment:
oh, this is not the first place i see the uaa code not honoring interffaces, and actually expecting actual impl classes ... with this it is hard to switch to using just jar ...